lesson17/src/js/main.js

```diff
@@ -0,0 +1,183 @@
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = createElementWithClassName("div", "slider__pagination");
+
+  const fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.id = 'js-current';
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createElementWithClassName("span", "separation");
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = 'js-total';
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function switchPagination(target) {
+  const pagenationCurrent = document.getElementById("js-current");
+  const pagenationTotal = Number(document.getElementById("js-total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  const buttonPrev = target.getAttribute("aria-label") === 'previous';
+  const buttonNext = target.getAttribute("aria-label") === 'next';
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum +=1;
+  }else if(currentNum <= pagenationTotal && buttonPrev){
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  active[target].classList.add("is-active");
+}
+
+function toggleButtonDisabled(){
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const activeImage = document.querySelector(".is-active");
+  const currentIndex = Number(activeImage.dataset.slideIndex);
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  prevButton.disabled = currentIndex === firstIndex;
+  nextButton.disabled = currentIndex === lastIndex;
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(fetchImageData("https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&delay=1000"));
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function callSlideContents() {
+  showLoadingImage();
+  try {
+    const json = await callImageData();
+    const slideImageData = json.slide;
+
+    renderSlideImage(slideImageData);
+    renderPagenation(slideImageData.length);
+    toggleButtonDisabled()
+  } finally {
+    removeLoading();
+  }
+}
+
+function init() {
+  renderSlideArea();
+  renderSlideButton();
+  callSlideContents();
+}
+
+init();
+
```

Comment:
```suggestion
async function init() {
  showLoadingImage();
  let slideImageData;
  try {
    const json = await callImageData();
    slideImageData = json.slide;
  } finally {
    removeLoading();
  }
  renderSlideImage(slideImageData);
  renderPagenation(slideImageData.length);
  renderSlideArea();
  renderSlideButton();
  toggleButtonDisabled();
}
init();
```
I am concerned that the callSlideContents and init contents are separated even though they are the same render**.
I am sharing this because I was pointed out to me in issue 17.
I think that only the data acquisition should be performed within the try.
I don't think it is necessary to go as far as render** within a try, and there is a risk of losing track when an error occurs.
It is easier to read if you put it all together in init and run render** together outside of try.

Reply:
Thank you for your review!
Based on the content of the review, we have separated the data acquisition function and the slide creation function.

I thought that it might be something that exchanges data within init, so I divided the functions.

[04616f1](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/04616f15326a0924c5290d2666c3f9b6c75713e2)

Reply:
個人的には関数が別れていない方が読みやすいと思いました。

> I thought that it might be something that exchanges data within init, so I divided the functions.

init内でデータを交換するものとはどういった意味でしょうか？理解不足で申し訳ないです🙏

Reply:
I'm sorry, it's late.
Image calls are put together in the init function.
Also added the processing when the number of images was 0.

[6719e63](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/6719e6344692587bc53eea965652939c8a9cf5b3)